### DISCUSSION TITLE
How WinGet interact with Windows Terminal

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/winget-cli#3375] [General]
Unfortunately, I didn't know what to search so that Google can find... \
I'm curious generally... maybe I will use it in the future... \
when I use the WinGet command I can see it interacts with Windows Terminal, Terminal shows a loading circle in the tab bar or loading line under the Windows Terminal icon in the Windows taskbar, How WinGet interact with Windows Terminal?

**Comment by florelis** [comment]
AFAIK, winget outputs some special escape sequences that the terminal intercepts and uses to set the progress. I know that's what's used for the spinner in the progress bar, to delete something we already wrote and put a new character there. Those are documented at https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences, but I didn't see anything about the progress specifically.

For search terms, I think it's based on something called "conemu", but I don't know what the details.
